packages/discord.js/typings/index.d.ts

```diff
@@ -2453,10 +2453,27 @@ export class Webhook extends WebhookMixin() {
   public sourceChannel: NewsChannel | APIPartialChannel | null;
   public token: string | null;
   public type: WebhookType;
-  public isIncoming(): this is this & { token: string };
+  public applicationId: Snowflake | null;
+  public isUserCreated(): this is this & {
+    type: WebhookType.Incoming;
+    applicationId: null;
+    owner: User | APIUser;
+    token: null;
+  };
+  public isApplicationCreated(): this is this & {
+    type: WebhookType.Incoming;
+    applicationId: Snowflake;
+    owner: null;
+  };
+  public isIncoming(): this is this & {
+    type: WebhookType.Incoming;
+    token: string;
+  };
   public isChannelFollower(): this is this & {
+    type: WebhookType.ChannelFollower;
     sourceGuild: Guild | APIPartialGuild;
     sourceChannel: NewsChannel | APIPartialChannel;
+    token: null;
   };
```

Comment:
The owner will be defined here! Also, `applicationId` will be `null` here, even if added by a bot.

Reply:
no, bot-created webhooks (as well as oauth2-created webhooks) have an application id

Reply:
The example on the docs shows that application webhooks can have an application id
https://discord.com/developers/docs/resources/webhook#webhook-object-example-application-webhook

Reply:
Seems that Jiralite was correct. `application_id` is null on following webhooks